clap_derive/src/utils/doc_comments.rs

```diff
@@ -117,10 +83,200 @@ fn is_blank(s: &str) -> bool {
     s.trim().is_empty()
 }
 
-fn merge_lines(lines: impl IntoIterator<Item = impl AsRef<str>>) -> String {
-    lines
-        .into_iter()
-        .map(|s| s.as_ref().trim().to_owned())
-        .collect::<Vec<_>>()
-        .join(" ")
+mod markdown {
+    use anstyle::{Reset, Style};
+    use pulldown_cmark::{Event, Options, Parser, Tag, TagEnd};
+    use std::fmt;
+    use std::fmt::Write;
+    use std::ops::AddAssign;
+
+    #[derive(Default)]
+    struct MarkdownWriter {
+        output: String,
+        indentation: usize,
+        styles: Vec<Style>,
+    }
+
+    impl MarkdownWriter {
+        fn newline(&mut self) {
+            self.output.push('\n');
+        }
+        fn endline(&mut self) {
+            if !self.output.ends_with('\n') {
+                self.newline();
+            }
+        }
+
+        fn write_fmt(&mut self, arguments: fmt::Arguments<'_>) {
+            if self.output.ends_with('\n') {
+                write!(self.output, "{0: <1$}", "", self.indentation).unwrap();
+            }
+            self.output.write_fmt(arguments).unwrap();
+        }
```

Comment:
On newlines, we should reset the output and start it up again on the next line

See #4767

Reply:
Probably not an issue as markdown styles should all be inline i.e., the reset should have already happened at the newline, but nevertheless better safe than sorry.

Reply:
This prompted me to test `wrap_help` and that seems to actually have a similar problem:

![grafik](https://github.com/user-attachments/assets/eef28817-004c-47de-b9b6-daadf38fcd38)